File: apps/scheduler/src/repository/schemas/judging_session.py
from typing import TypedDict, Literal
from datetime import datetime
from bson import ObjectId


class JudgingSession(TypedDict):
    divisionId: ObjectId
    number: int
    teamId: ObjectId | None
    roomId: ObjectId
    called: Literal[False]
    queued: Literal[False]
    status: Literal["not-started"]
    scheduledTime: datetime

File: libs/types/src/lib/old/schemas/judging-session.ts
import { ObjectId } from 'mongodb';
import { Status } from '../constants';

export interface JudgingSession {
  divisionId: ObjectId;
  number: number;
  teamId: ObjectId | null;
  roomId: ObjectId;
  called: boolean;
  queued: boolean;
  status: Status;
  scheduledTime: Date;
  startTime?: Date;
}
